docs/workflows/input_output.md

```diff
@@ -16,12 +16,50 @@ The Cumulus Message Adapter and Cumulus Message Adapter libraries help task deve
 
 The Cumulus Message Adapter libraries are called by the tasks with a callback function containing the business logic of the task as a parameter. They first adapt the incoming message to a format more easily consumable by Cumulus tasks, then invoke the task, and then adapt the task response back to the Cumulus message protocol to be sent to the next task.
 
-A task's Lambda function can be configured to include a Cumulus Message Adapter library which constructs input/output messages and resolves task configurations. In the Lambda function configuration file lambdas.yml, a task Lambda function can be configured to use Cumulus Message Adapter, for example:
+A task's Lambda function can be configured to include a Cumulus Message Adapter library which constructs input/output messages and resolves task configurations.     The CMA can then be included in one of three ways:
+
+
+#### Kes injection
+
+In the Lambda function configuration file lambdas.yml, a task Lambda function can be configured to include the latest CMA via kes.  Kes will download and include the latest CMA package in the lambda that's uploaded to AWS:
 
     DiscoverPdrs:
       handler: index.handler
       useMessageAdapter: true
 
+#### Lambda Layer
+
+In order to make use of this configuration, a lambda layer can be uploaded to your account.  Due to platform restrictions, Core cannot currently support sharable public layers, however you can support deploying the appropriate version from [the release page](https://github.com/nasa/cumulus-message-adapter/releases) via the AWS [Layers Interface](https://console.aws.amazon.com/lambda/home?region=us-east-1#/layers), *or* the provided CMA [Terraform](https://www.terraform.io/) module located at [tf-modules/cumulus-message-adapter](https://github.com/nasa/cumulus/tree/master/tf-modules/cumulus-message-adapter).
+
+Once you've deployed the layer, include the CMA in the configured lambda layers:
+
+    DiscoverPdrs:
+      layers:
+        - arn:aws:lambda:us-east-1:{{AWS_ACCOUNT_ID}}:layer:Cumulus_Message_Adapter:{version number}
+
```

Comment:
Is there a reason for the extra lines? Incidentally, is it possible to install a markdown linting tool for your editor? I ask because we do have a markdown linting config in the repo that would catch a lot of these issues

Reply:
I do, but it didn't catch said error.   I think we should probably discuss adding it to the CI linting step if we're intent on enforcing it, that should be pretty trivial.

Reply:
38e8c73d2